src/bio/bio_xstream.c

```diff
@@ -160,6 +210,22 @@ populate_whitelist(struct spdk_env_opts *opts)
 			break;
 		}
 
+		if (vmd_enabled) {
+			/*
+			 * TODO: Remove duplicate VMD addr being added to
+			 * whitelist for multiple NVMe SSDs behind the same VMD.
```

Comment:
That's something I'm not quite clear so far. What's the whitelist for? Is it ok to pass same VMD address to two IO server's whitelist? (when two io servers use different devices behind the same VMD)

Reply:
I was under the impression that the same VMD address wouldn't be used on multiple IO servers, @sydneyvanda is that true? if the VMD address to be shared , then we can't use it in the whitelist

Reply:
The VMD address is server/node specific, like PCIe address. The TODO here is removing the additional address when there are 2 NVMe SSDs behind the same address (ie it will do something like "DPDK EAL parameters: daos --pci-whitelist=0000:5d:05.5 --pci-whitelist=0000:5d:05.5" ). Still thinking of a creative way to get it to skip if the same address is used. Nothing seems to happen when passing the duplicate so far in my testing, I'm thinking DPDK just ignores it, however it is just redundant on the console output.